main.js

```diff
@@ -160,4 +160,152 @@ const inventory = [
+  let count = 0
+  for (let i = 0; i < televisions.length; i++) {
+    const revenueGoal = televisions[i].price * televisions[i].originalStock
+    count = count + revenueGoal
+  }
+  return count
+}
+//Implementation of above function on website
+const revenueCount = calculateRevenueGoal(inventory)
+const revenueCountSite = document.getElementById('optimalRevenue')
+revenueCountSite.textContent = "€ " + revenueCount.toString()
+//Function to calculate revenue until now
+function calculateActualRevenue(televisions) {
+  let count = 0
+  for (let i = 0; i < televisions.length; i++) {
+    const actualRevenue = televisions[i].price * televisions[i].sold
+    count = count + actualRevenue
+  }
+  return count
+}
+//Implementation of above function on website
+const actualRevenueCount = calculateActualRevenue(inventory)
+const actualRevenueCountSite = document.getElementById('actualRevenue')
+actualRevenueCountSite.textContent = "€ " + actualRevenueCount.toString()
+//Function to return TV name & type in string
+function generateStringTvName(television) {
+  return television.brand + " " + television.type + ' - ' + television.name
+}
+//Function to return price with euro sign
+function generateTVprice(television) {
+  return '€'+television.price+',-'
+}
+//Function to display screen sizes in inch & cm
+function getScreenSizes(television) {
+  let tvSizesArrayInch = television.availableSizes
+  let tvSizeCM = []
+  let tvSizeInch = []
+  for (let i = 0; i < tvSizesArrayInch.length; i++) {
+      tvSizeInch.push(tvSizesArrayInch[i])
+      const sizeRoundedCm = Math.round(tvSizesArrayInch[i] * 2.54)
+      tvSizeCM.push(tvSizeInch[i] + ' inch (' + sizeRoundedCm + ' cm) |  ' )
+  }
+  const tvSizeJoined = tvSizeCM.join("")
+  return tvSizeJoined.toString()
+}
+//Function to display one television out of array
+function displayOneTelevision(television) {
+  return generateStringTvName(television) + '\n' + generateTVprice(television) + '\n' + getScreenSizes(television)
+}
+//Function to display complete array of televisions
+function displayAllTelevisions(televisions) {
+  let printThis = "";
+  for (let i = 0; i < televisions.length; i++) {
+    printThis += '<b>' + generateStringTvName(televisions[i]) + '</b>' + '<br>' + generateTVprice(televisions[i]) + '<br>' + getScreenSizes(televisions[i]) + '<br><br>'
+  }
+  return printThis
+}
+//Implementation of above function on website
+const parent = document.getElementById('showTV')
+parent.innerHTML = (displayAllTelevisions(inventory))
+//Function to display TV's sorted by price low to high
+function displayAllTelevisionsSorted(televisions) {
+  const sortTvByPrice = televisions.sort((a, b) => {
+    return a.price - b.price })
+  let printThis = "";
+  for (let i = 0; i < televisions.length; i++) {
+    printThis += '<b>' + generateStringTvName(televisions[i]) + '</b>' + '<br>' + generateTVprice(televisions[i]) + '<br>' + getScreenSizes(televisions[i]) + '<br><br>'
+  }
+  return printThis
```

Comment:
Wat je hier had kunnen doen is eerst inventory sorteren a.d.h.v. de prijs en dan de `displayAllTelevisions` functie aan kunnen roepen. Op deze manier heb je geen code dubbeling. Ik tel nu 4x dezelfde soort code, wat onderhoudbaarheid niet ten goede komt.

```
televisions.sort((a,b) => a.price - b.price);
displayAllTelevisions(inventory);
```